notebook/app.py

```diff
@@ -62,6 +62,9 @@ def get_page_config(self):
             api_token = os.getenv("JUPYTERHUB_API_TOKEN", "")
             page_config["token"] = api_token
 
+        nbclassic_enabled = self.server_extension_is_enabled("nbclassic")
+        page_config["nbclassic_enabled"] = nbclassic_enabled
```

Comment:
We should check if there is a way to set some default page config options when the notebook server app is initialized.

This could be placed at the beginning of `initialize_handlers`:

https://github.com/jupyter/notebook/blob/245bc87ecf56b5a86be0595bc86f709110340859/notebook/app.py#L264-L279

For reference JupyterLab sets some default page config options as part of `initialize_handlers`:

https://github.com/jupyterlab/jupyterlab/blob/6444459940395343a9999a40af43b8b621cc6ace/jupyterlab/labapp.py#L691-L702

So if we update notebook to do something like the following, then JupyterLab should also be able to pick the `nbclassic_enabled` option in its page config:

```py
def initialize_handlers(self):
        """Initialize handlers."""
       nbclassic_enabled = self.server_extension_is_enabled("nbclassic")
       page_config["nbclassic_enabled"] = nbclassic_enabled
       self.serverapp.web_app.settings["page_config_data"] = page_config
```

Reply:
Which means the server extension (added in this PR) would not be needed anymore.

Reply:
Setting page_config in initialize_handlers works great, thank you for the suggestion

Reply:
![Open_in](https://github.com/jupyter/notebook/assets/26686070/835f05b3-a572-496e-933b-e2b68beccdc3)